src-plugins/medSegmentation/msegAlgorithmPaintToolbox.cpp

```diff
@@ -760,6 +696,7 @@ void AlgorithmPaintToolbox::clearMask()
         {
             m_imageData->removeAttachedData(m_maskAnnotationData);
             maskHasBeenSaved = false;
+            setData(currentView->layerData(0));
```

Comment:
Origin of my "let's get rid of multi-layer management". I couldn't use currentView->currentLayer() because the current layer was actually the mask, which is about to be deleted. So the best/simplest way was to constrain the data to be on the first layer.